crates/resolver/src/hosts.rs

```diff
@@ -35,10 +35,17 @@ impl Hosts {
     /// Creates a new configuration from the system hosts file,
     /// only works for Windows and Unix-like OSes,
     /// will return empty configuration on others
+    #[cfg(any(unix, windows))]
     pub fn new() -> Self {
         read_hosts_conf(hosts_path()).unwrap_or_default()
     }
 
+    /// Creates a default configuration for non Windows or Unix-like OSes
+    #[cfg(not(any(unix, windows)))]
+    pub fn new() -> Self {
```

Comment:
Not sure about this one. For this use case, callers can just use `Hosts::default()` directly?

Reply:
I wasn't sure about this either, but I was just following what the doc comment says above `new()`.

Reply:
Sure, I meant -- if the non-`unix, windows` implementation is just going to call `Hosts::default()` which is already public API, I think it'd be clearer to make `new()` a compile error on non-`unix, windows`.

Reply:
I see. I will remove this then and leave the guard over the already public `new()`? 
`hosts_path()` is not available on non-unix,windows

Reply:
Yes, sounds good!

Reply:
Actually, due to https://github.com/bluejekyll/trust-dns/blob/11b84140ff4ec7cf4c7c5a541579697da1b1d724/crates/resolver/src/async_resolver.rs#L239 I think `new()` needs to be present. Any other ideas?

Reply:
Actually, I'm wondering if we should just leave the `new()` because it is used here: https://github.com/bluejekyll/trust-dns/blob/11b84140ff4ec7cf4c7c5a541579697da1b1d724/crates/resolver/src/async_resolver.rs#L239

I suppose I could add a guard around that line. Thoughts?

Reply:
Okay, yeah, maybe easier to just leave it then.